homeassistant/components/sensor/mqtt.py

```diff
@@ -74,15 +82,36 @@ def async_added_to_hass(self):
         @callback
         def message_received(topic, payload, qos):
             """A new MQTT message has been received."""
+            # auto-expire enabled?
+            if self._expire_after > 0:
+                # Reset expiration time and set trigger
+                self._value_expiration_at = (
+                    dt_util.utcnow() +
+                    timedelta(seconds=self._expire_after)
+                    )
+                async_track_point_in_utc_time(self.hass,
```

Comment:
Will this be thread-safe? E.g. no race-condition between unsubscribing a (probably in exactly in this moment executed callback) and a new message?

Flow with new code:
- Callback is executed in the moment the message comes in
- Unsubscribe is executed -> too late to have effect
- new value is set
- already running Callback resets the value

The more I think about it, the more I have the impression that I can also run in that with my current code:
Flow with my old code:
- Callback is executed in the moment the message comes in
- Callback checks time, detects that the value is to expire
- new expire time and value is set
- already running Callback resets the value

Any thoughts about the thread safety here?